packages/core/src/modules/proofs/services/ProofService.ts

```diff
@@ -360,6 +366,9 @@ export class ProofService {
     }
     await validateOrReject(proofRequest)
 
+    // Check if Attribute names match Predicate names
+    this.checkAttributePredicateMatch(proofRequest)
```

Comment:
This comment makes me believe we want them to match. 

Maybe something like:

```suggestion
    // Assert attribute and predicate group names do not match
    this.assertAttributePredicateGroupNamesDoNotMatch(proofRequest)
```

Not sure if doNotMatch is the best, but is is explicit